src/CacheManager.SystemRuntimeCaching/MemoryCacheHandle`1.cs

```diff
@@ -70,6 +73,12 @@ public MemoryCacheHandle(ICacheManagerConfiguration managerConfiguration, CacheH
         public override int Count => (int)_cache.GetCount();
 
         /// <inheritdoc />
+        protected override IEnumerable<string> AllKeys()
+        {
+            return _cache.Select(c => c.Key);
```

Comment:
You'd have to retrurn only the keys starting with `instancekey`. Depending on if a region exists, too, the separators might be different